homeassistant/helpers/config_validation.py

```diff
@@ -65,9 +66,15 @@ def boolean(value: Any) -> bool:
     return bool(value)
 
 
-def isfile(value):
+def isfile(value: Any) -> str:
     """Validate that the value is an existing file."""
-    return vol.IsFile('not a file')(value)
+    if value is None:
+        raise vol.Invalid('None is not file')
+    file_in = str(value)
+
+    if not os.path.isfile(file_in) or not os.access(file_in, os.R_OK):
+        raise vol.Invalid('not a file')
```

Comment:
Can we split this if-check into two if-checks so that we can raise exceptions with correct message (Not a file & File not readable)